### PR TITLE
add msg title and body in template

### PR DESCRIPTION
**Pull request by zwpaper**
add `build.messageTitle` and `build.messageBody` for the commit message first line and rest

fix https://github.com/drone-plugins/drone-slack/issues/56

### CODE REVIEW

**Comment by zwpaper** [comment]
1. git will not allow an empty commit msg
2. Even an empty msg, split will return one empty element

So it will never panic

**Comment by zwpaper** [comment]
For readability, I added a comment for why not checking the length

**Comment by tboerger** [comment]
> 2\. Even an empty msg, split will return one empty element

Yeah, one element, but for the commit message body you are starting after the first element.

**Comment by zwpaper** [comment]
> starting after the first element.

You mean the msg[1:], but this will work perfectly with length 1.

You can check here
https://play.golang.org/p/7RxSbGd4DNh

Without doing this, I have to write some dead code, checking if length less than 1, which will never be, use a empty string

**Comment by lafriks** [comment]
Please resolve conflicts

**Comment by zwpaper** [comment]
@tboerger can you help to ask someone reviewing the code, or I have to resolve the conflicts next time...

**Comment by tboerger** [comment]
I know that I already approved the pr, but could you still create a message struct with title and body attribute and add a string function to it the keeps the current behavior? Would be a better structure and still keep the default behavior.

**Comment by zwpaper** [comment]
@tboerger forgive me for the delay, added message struct and some tests for newly added codes, review it please, correct me if errors

File: plugin.go [hunk 1]
```diff
@@ -66,9 +72,24 @@ func (a Author) String() string {
...
+	splitMsg := strings.Split(m, "\n")
+
+	return Message{
+		msg:   m,
+		Title: strings.TrimSpace(splitMsg[0]),
+		Body:  strings.TrimSpace(strings.Join(splitMsg[1:], "\n")),
+	}
+}
+func (m Message) String() string {
+	return m.msg
```
Comment (lafriks) [inline plugin.go]: Can't this return just title + newline + body?

Comment (zwpaper) [reply plugin.go]: there are some commit messages using an empty second line.

if we do in that way, we will lose the empty second line,
as the Message.msg is not high cost, so IMO, using it for saving the origin format is worthy.

**Pull request merged**